tests/integration-tests/tests/cli_commands/test_cli_commands.py

```diff
@@ -19,43 +20,240 @@
+    )
+    cluster = _test_create_cluster(clusters_factory, cluster_config, cluster_config_with_warning, request)
     remote_command_executor = RemoteCommandExecutor(cluster)
+    _test_list_cluster(cluster.name, "CREATE_COMPLETE")
 
-    instance_ids = _test_pcluster_instances_and_status(cluster, region, compute_fleet_status="RUNNING")
-    _test_pcluster_stop_and_start(cluster, region, expected_num_nodes=2)
+    instance_ids = _test_describe_instances(cluster, region)
+    check_status(cluster, "CREATE_COMPLETE", "running", "RUNNING")
+    _test_describe_instances(cluster, region, node_type="HeadNode")
+    _test_describe_instances(cluster, region, node_type="Compute")
+    _test_describe_instances(cluster, region, queue_name="ondemand1")
+    _test_pcluster_compute_fleet(cluster, region, expected_num_nodes=2)
     _test_pcluster_export_cluster_logs(s3_bucket_factory, cluster, region, instance_ids)
     cfn_init_log_stream = _test_pcluster_list_cluster_logs(cluster, instance_ids)
     _test_pcluster_get_cluster_log_events(cluster, cfn_init_log_stream)
     assert_no_errors_in_logs(remote_command_executor, scheduler)
 
 
-def _test_pcluster_instances_and_status(cluster, region, compute_fleet_status=None):
-    """Test pcluster status and pcluster instances functionalities."""
-    logging.info("Testing that pcluster status and pcluster instances output are expected")
-    cluster_instances_from_ec2 = get_cluster_nodes_instance_ids(cluster.cfn_name, region)
-    cluster_instances_from_cli = cluster.instances()
+def _test_create_cluster(clusters_factory, cluster_config, cluster_config_with_warning, request):
+    if not request.config.getoption("cluster"):
+        # Test below is not compatible with `--cluster` flag. Therefore, skip it if the flag is provided.
+        logging.info("Testing cluster creation on a configuration file with 1 error and 2 warnings.")
+        custom_ami_tag_warning = {
+            "level": "WARNING",
+            "type": "CustomAmiTagValidator",
+            "message": "The custom AMI may not.*been created by pcluster",
+        }
+        name_error = {
+            "level": "ERROR",
+            "type": "NameValidator",
+            "message": "Name must begin with a letter and only contain lowercase letters, digits and hyphens.",
+        }
+        key_pair_warning = {
+            "level": "WARNING",
+            "type": "KeyPairValidator",
+            "message": ".*you do not specify a key pair.*",
+        }
+        invalid_cluster_configuration_message = "Invalid cluster configuration"
+        expected_creation_response = {
+            "configurationValidationErrors": [custom_ami_tag_warning, key_pair_warning, name_error],
+            "message": invalid_cluster_configuration_message,
+        }
+        # Set log_error=False if error is expected. Therefore, the test log won't be flooded with errors
+        cluster = clusters_factory(
```

Comment:
It looks like there's a decent amount of code below here that is structurally the same... I'm curious if we can turn this into a more data-based approach where we have a list of args and a list of expected responses and then iterate through each of them performing the same:
```
    logging.info(f"Testing {test_description}")
    cluster = clusters_factory(cluster_config_with_warning, extra_args, ...)
    _check_response_with_configuration_validation(cluster.creation_response, expected_creation_response)
```

This would put the inputs next to the outputs and only have the actual operations in a single place.

Reply:
Done